audit-processors/src/main/java/uk/gov/di/authentication/audit/lambda/CounterFraudAuditReplayLambda.java

```diff
@@ -0,0 +1,108 @@
+package uk.gov.di.authentication.audit.lambda;
+
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.S3Event;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ObjectMessage;
+import uk.gov.di.audit.AuditPayload.AuditEvent;
+import uk.gov.di.audit.AuditPayload.AuditEvent.User;
+import uk.gov.di.authentication.shared.services.ConfigurationService;
+
+import java.util.HashMap;
+
+import static org.apache.commons.codec.binary.Hex.encodeHexString;
+import static uk.gov.di.authentication.audit.helper.HmacSha256Helper.hmacSha256;
+
+public class CounterFraudAuditReplayLambda implements RequestHandler<S3Event, Void> {
+
+    private static final Logger LOG = LogManager.getLogger(CounterFraudAuditReplayLambda.class);
+
+    private final ConfigurationService configurationService;
+    private final String hmacKey;
+    private final AmazonS3 client;
+
+    public CounterFraudAuditReplayLambda(
+            ConfigurationService configurationService, AmazonS3 client) {
+        this.configurationService = configurationService;
+        this.hmacKey = configurationService.getAuditHmacSecret();
+        this.client = client;
+    }
+
+    public CounterFraudAuditReplayLambda() {
+        this(
+                ConfigurationService.getInstance(),
+                AmazonS3Client.builder().withRegion(Regions.EU_WEST_2).build());
+    }
+
+    @Override
+    public Void handleRequest(S3Event input, Context context) {
+
+        input.getRecords()
+                .forEach(
+                        s3 -> {
+                            var key = s3.getS3().getObject().getKey();
+                            var bucket = s3.getS3().getBucket().getName();
+
+                            LOG.info("Processing file {}", key);
+                            var content = client.getObjectAsString(bucket, key);
```

Comment:
We're assuming that each file contains 1 event here, I'm not sure if that's 100% true?